homeassistant/components/ping/binary_sensor.py

```diff
@@ -226,7 +226,7 @@ async def async_ping(self):
         )
         try:
             out_data, out_error = await asyncio.wait_for(
-                pinger.communicate(), self._count + PING_TIMEOUT
+                pinger.communicate(), self._count + ICMP_TIMEOUT + PING_TIMEOUT
```

Comment:
the `ping` command sends every 1s an ICMP echorequest (_Independent of given timeout value_)